{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e585ae14",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error accessing website http://www.doggroomingparlour.co.uk: 403 Client Error: Forbidden for url: http://www.doggroomingparlour.co.uk/\n",
      "Emails extracted for http://www.doggiestylemobileunit.co.uk: \n",
      "Emails extracted for https://www.awesomepaws.org/: pageinfo@awesomepaws.orgLog\n",
      "Error accessing website http://www.pawsawhiledoggrooming.co.uk: HTTPConnectionPool(host='www.pawsawhiledoggrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5C9B2BF0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for http://www.lolasdoggrooming.com: lolasdoggrooming@outlook.com\n",
      "Emails extracted for http://www.paw-pals.co.uk: \n",
      "Emails extracted for https://www.facebook.com/norton-dog-groomers-266156213418722/: \n",
      "Emails extracted for http://www.caninedesignstockton.co.uk: \n",
      "Emails extracted for https://www.facebook.com/thedogroomstudio: \n",
      "Emails extracted for http://www.alphadoggrooming.yolasite.com: \n",
      "Emails extracted for http://www.thebarkinglotltd.co.uk: \n",
      "Emails extracted for http://www.pamperedpupz.co.uk/: \n",
      "Emails extracted for http://www.inglebypets.co.uk: \n",
      "Emails extracted for http://www.sherbetdoggrooming.co.uk: \n",
      "Emails extracted for http://www.jollyes.co.uk: \n",
      "Error accessing website https://inglebypets.co.uk: HTTPSConnectionPool(host='inglebypets.co.uk', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000029F5DEE94E0>, 'Connection to inglebypets.co.uk timed out. (connect timeout=None)'))\n",
      "Emails extracted for https://community.petsathome.com/store/teesside/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2Njc5NjctNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Error accessing website http://www.amyspets.co.uk: HTTPConnectionPool(host='www.amyspets.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E024CA0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.furcutzbillingham.com/: \n",
      "Emails extracted for https://www.facebook.com/pages/category/pet-service/doggie-chops-2098181433827294: \n",
      "Error accessing website http://www.woofnreadypetgrooming.co.uk: 403 Client Error: Forbidden for url: http://www.woofnreadypetgrooming.co.uk/\n",
      "Error accessing website https://callywaggiesdoggrooming.co.uk: HTTPSConnectionPool(host='callywaggiesdoggrooming.co.uk', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:997)')))\n",
      "Emails extracted for https://www.facebook.com/thebarkingl/: \n",
      "Error accessing website http://www.doggiedaysteesside.com: 403 Client Error: Forbidden for url: http://www.doggiedaysteesside.com/\n",
      "Error accessing website http://www.louisasdoggroomingsalon.co.uk: 403 Client Error: Forbidden for url: http://www.louisasdoggroomingsalon.co.uk/\n",
      "Emails extracted for http://www.dhgrooming.com: \n",
      "Emails extracted for https://www.facebook.com/zibizipet/: \n",
      "Emails extracted for http://www.parkvalepetgrooming.co.uk: GROOMER@PARKVALEPETGROOMING.CO.UK\n",
      "Emails extracted for https://www.facebook.com/bark-in-style-431251204296802/: \n",
      "Emails extracted for http://www.vanity-fur-grooming.com: info@vanity-fur-grooming.com\n",
      "Error accessing website http://www.k9houseofstyle.com: 404 Client Error: Not Found for url: https://www.k9houseofstyle.com/\n",
      "Error accessing website http://www.barkbone.com: 403 Client Error: Forbidden for url: https://www.brandbucket.com/names/barkbone?source=ext\n",
      "Emails extracted for https://www.facebook.com/solodoggrooming/: \n",
      "Emails extracted for https://www.facebook.com/profile.php?id=100083123357268: \n",
      "Error accessing website http://www.place4pets.co.uk: HTTPConnectionPool(host='www.place4pets.co.uk', port=80): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x0000029F5E2A8490>, 'Connection to www.place4pets.co.uk timed out. (connect timeout=None)'))\n",
      "Emails extracted for https://www.facebook.com/waggywandererspetcare/: \n",
      "Emails extracted for http://www.bartlebridgekennels.co.uk: \n",
      "Emails extracted for https://community.petsathome.com/groom-room/middlesbrough/?y_source=1_MjMyNjI4MzAtNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for http://www.thefairydogmother.co.uk/: \n",
      "Emails extracted for https://www.facebook.com/pages/category/pet-groomer/top-dog-groomers-darlington-372176793350756: \n",
      "Emails extracted for https://www.facebook.com/stalbansdoggrooming: \n",
      "Emails extracted for http://www.breedabove.co.uk: info@breedabove.co.uk, info@breedabove.co.uk\n",
      "Emails extracted for https://community.petsathome.com/store/coulby-newham/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2Njc4OTItNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for https://viccis-doggy-spa.business.site/?utm_source=yell&utm_medium=referral&utm_campaign=yell: \n",
      "Emails extracted for http://www.pedigreecuts.co.uk: \n",
      "Emails extracted for https://www.facebook.com/doggydaycutz: \n",
      "Error accessing website https://www.woofsandwags.info/: HTTPSConnectionPool(host='www.woofsandwags.info', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5E11B3D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.facebook.com/trishsprofessionaldoggrooming: \n",
      "Error accessing website http://www.tomstrims.co.uk: HTTPConnectionPool(host='www.tomstrims.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E14A9E0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://community.petsathome.com/groom-room/darlington/?y_source=1_Mjc5OTY3ODQtNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for https://www.facebook.com/prettypawfectdarlington: \n",
      "Emails extracted for https://www.facebook.com/sprucedupstokesley: \n",
      "Emails extracted for http://www.okinastar.co.uk/: \n",
      "Emails extracted for https://www.facebook.com/diamondkutzmobiledoggrooming: \n",
      "Emails extracted for https://www.facebook.com/claws2paw/: \n",
      "Emails extracted for http://www.recommendhartlepool.co.uk: \n",
      "Emails extracted for https://community.petsathome.com/store/middlesbrough/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2Njc4OTktNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for http://www.shop.pawfect.supplies: \n",
      "Emails extracted for http://www.dogzwash.co.uk: \n",
      "Emails extracted for http://www.snip-it.uk/: \n",
      "Emails extracted for https://community.petsathome.com/store/darlington/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2Njc3MDAtNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for https://community.petsathome.com/store/hartlepool/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2Njc5MjMtNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Error accessing website http://www.whiskerspawsandchores.co.uk: HTTPConnectionPool(host='www.whiskerspawsandchores.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5DC5E8F0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.sageprofessionaldoggrooming.co.uk/: 650711info@sageprofessionaldoggrooming.co.ukHomeDog, touchinfo@sageprofessionaldoggrooming.co.ukHighly, 6RFinfo@sageprofessionaldoggrooming.co\n",
      "Emails extracted for http://www.comfyk9s.co.uk: \n",
      "Emails extracted for https://www.facebook.com/ScruffyPupsCastletown/: \n",
      "Error accessing website http://www.kingsroadvets.org?utm_source=yell&utm_medium=referral&utm_campaign=yell: 406 Client Error: Not Acceptable for url: http://www.kingsroadvets.org/?utm_source=yell&utm_medium=referral&utm_campaign=yell\n",
      "Emails extracted for https://www.pinkwhiskersgrooming.co.uk/: info@pinkwhiskersgrooming.co.uk\n",
      "Error accessing website http://www.groomer-has-it.com/: HTTPConnectionPool(host='www.groomer-has-it.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E0E49D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error accessing website http://www.prettypawfectdoggrooming.co.uk: HTTPConnectionPool(host='www.prettypawfectdoggrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E0E4AF0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.facebook.com/pages/classy-cutie-cutz-dog-grooming/649015185111621: \n",
      "Emails extracted for https://www.facebook.com/thegroomingshedsunderland/: \n",
      "Error accessing website http://www.thedogpamperer.net: 404 Client Error: Not Found for url: http://www.thedogpamperer.net/\n",
      "Emails extracted for http://www.millionhairs.co.uk/: enquiries@millionhairs.co.uk, chris@arcimedia.co.uk, enquiries@millionhairs.co.uk\n",
      "Emails extracted for https://www.facebook.com/61frontstreet: \n",
      "Emails extracted for http://www.simplypawsome.co.uk: \n",
      "Emails extracted for https://www.facebook.com/people/Robsons-Dog-Grooming/100085000148978/: \n",
      "Emails extracted for http://www.darcysgroomingsalon.com: \n",
      "Emails extracted for https://www.facebook.com/k9groominggala: \n",
      "Emails extracted for http://www.stoneygatestableskennelsandcattery.co.uk: \n",
      "Emails extracted for https://community.petsathome.com/store/sunderland/?utm_source=ExtNet&utm_medium=Yext&y_source=1_MTU2NjgwMDQtNzAyLWxvY2F0aW9uLndlYnNpdGU%3D: \n",
      "Emails extracted for http://www.poshpawsdoggroomers.co.uk: \n",
      "Error accessing website http://www.binxsgroomingempawrium.business.site: HTTPSConnectionPool(host='www.binxsgroomingempawrium.business.site', port=443): Max retries exceeded with url: / (Caused by SSLError(CertificateError(\"hostname 'www.binxsgroomingempawrium.business.site' doesn't match either of '*.google.com', '*.appengine.google.com', '*.bdn.dev', '*.origin-test.bdn.dev', '*.cloud.google.com', '*.crowdsource.google.com', '*.datacompute.google.com', '*.google.ca', '*.google.cl', '*.google.co.in', '*.google.co.jp', '*.google.co.uk', '*.google.com.ar', '*.google.com.au', '*.google.com.br', '*.google.com.co', '*.google.com.mx', '*.google.com.tr', '*.google.com.vn', '*.google.de', '*.google.es', '*.google.fr', '*.google.hu', '*.google.it', '*.google.nl', '*.google.pl', '*.google.pt', '*.googleadapis.com', '*.googleapis.cn', '*.googlevideo.com', '*.gstatic.cn', '*.gstatic-cn.com', 'googlecnapps.cn', '*.googlecnapps.cn', 'googleapps-cn.com', '*.googleapps-cn.com', 'gkecnapps.cn', '*.gkecnapps.cn', 'googledownloads.cn', '*.googledownloads.cn', 'recaptcha.net.cn', '*.recaptcha.net.cn', 'recaptcha-cn.net', '*.recaptcha-cn.net', 'widevine.cn', '*.widevine.cn', 'ampproject.org.cn', '*.ampproject.org.cn', 'ampproject.net.cn', '*.ampproject.net.cn', 'google-analytics-cn.com', '*.google-analytics-cn.com', 'googleadservices-cn.com', '*.googleadservices-cn.com', 'googlevads-cn.com', '*.googlevads-cn.com', 'googleapis-cn.com', '*.googleapis-cn.com', 'googleoptimize-cn.com', '*.googleoptimize-cn.com', 'doubleclick-cn.net', '*.doubleclick-cn.net', '*.fls.doubleclick-cn.net', '*.g.doubleclick-cn.net', 'doubleclick.cn', '*.doubleclick.cn', '*.fls.doubleclick.cn', '*.g.doubleclick.cn', 'dartsearch-cn.net', '*.dartsearch-cn.net', 'googletraveladservices-cn.com', '*.googletraveladservices-cn.com', 'googletagservices-cn.com', '*.googletagservices-cn.com', 'googletagmanager-cn.com', '*.googletagmanager-cn.com', 'googlesyndication-cn.com', '*.googlesyndication-cn.com', '*.safeframe.googlesyndication-cn.com', 'app-measurement-cn.com', '*.app-measurement-cn.com', 'gvt1-cn.com', '*.gvt1-cn.com', 'gvt2-cn.com', '*.gvt2-cn.com', '2mdn-cn.net', '*.2mdn-cn.net', 'googleflights-cn.net', '*.googleflights-cn.net', 'admob-cn.com', '*.admob-cn.com', 'googlesandbox-cn.com', '*.googlesandbox-cn.com', '*.safenup.googlesandbox-cn.com', '*.gstatic.com', '*.metric.gstatic.com', '*.gvt1.com', '*.gcpcdn.gvt1.com', '*.gvt2.com', '*.gcp.gvt2.com', '*.url.google.com', '*.youtube-nocookie.com', '*.ytimg.com', 'android.com', '*.android.com', '*.flash.android.com', 'g.cn', '*.g.cn', 'g.co', '*.g.co', 'goo.gl', 'www.goo.gl', 'google-analytics.com', '*.google-analytics.com', 'google.com', 'googlecommerce.com', '*.googlecommerce.com', 'ggpht.cn', '*.ggpht.cn', 'urchin.com', '*.urchin.com', 'youtu.be', 'youtube.com', '*.youtube.com', 'youtubeeducation.com', '*.youtubeeducation.com', 'youtubekids.com', '*.youtubekids.com', 'yt.be', '*.yt.be', 'android.clients.google.com', 'developer.android.google.cn', 'developers.android.google.cn', 'source.android.google.cn'\")))\n",
      "Emails extracted for http://www.the-grooming-room.co.uk: Groomingroommcl@gmail.com\n",
      "Emails extracted for http://www.petnurseservices.co.uk: \n",
      "Emails extracted for https://www.facebook.com/pages/category/pet-service/posh-dogs-grooming-salon-112934053692472: \n",
      "Emails extracted for https://www.facebook.com/kilopetgrooming: \n",
      "Emails extracted for https://trimwag.business.site/?utm_source=yell&utm_medium=referral&utm_campaign=yell: \n",
      "Emails extracted for https://www.facebook.com/chesterpetstores/: \n",
      "Error accessing website http://www.smartdogsgrooming.co.uk: HTTPConnectionPool(host='www.smartdogsgrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E354C40>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://facebook.com/alysonswags2whiskers: \n",
      "Emails extracted for https://www.facebook.com/k9clippersltdnewcastle: \n",
      "Emails extracted for http://www.tailsatseaham.co.uk: \n",
      "Error accessing website https://bellpetservices.co.uk/: 404 Client Error: Not Found for url: https://bellpetservices.co.uk/\n",
      "Emails extracted for http://www.scruffstocrufts.net: \n",
      "Emails extracted for https://www.facebook.com/pages/category/pet-groomer/top-dogs-grooming-studio-170882262930458: \n",
      "Emails extracted for https://www.facebook.com/scissorhounds/: \n",
      "Emails extracted for https://www.facebook.com/rachaelsmobiledoggrooming: \n",
      "Error accessing website http://www.a1prettypaws.com: HTTPConnectionPool(host='www.a1prettypaws.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5DEF0C70>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for http://www.petmeals.co.uk: \n",
      "Emails extracted for http://www.pawfectstyle.org.uk: enquiries@pawfectstyle.org.uk\n",
      "Emails extracted for https://www.facebook.com/LalasDogGroommingParlour/: \n",
      "Emails extracted for http://www.doras.dog: \n",
      "Emails extracted for http://www.dialadogwash.com: \n",
      "Emails extracted for http://www.pets2impress.com: mail@pets2impress.com\n",
      "Error accessing website http://www.bellasdoggrooming.co.uk/?utm_source=yell&utm_medium=referral&utm_campaign=yell: 403 Client Error: Forbidden for url: http://www.bellasdoggrooming.co.uk/?utm_source=yell&utm_medium=referral&utm_campaign=yell\n",
      "Emails extracted for https://www.facebook.com/houndsoflovegroomingsalon/: \n",
      "Emails extracted for http://www.mydogbarber.com?utm_source=yell&utm_medium=referral&utm_campaign=yell: \n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[2], line 22\u001b[0m\n\u001b[0;32m     18\u001b[0m website \u001b[38;5;241m=\u001b[39m sheet[website_column \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(row)]\u001b[38;5;241m.\u001b[39mvalue\n\u001b[0;32m     20\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m     21\u001b[0m     \u001b[38;5;66;03m# Send a GET request to the website\u001b[39;00m\n\u001b[1;32m---> 22\u001b[0m     response \u001b[38;5;241m=\u001b[39m \u001b[43mrequests\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwebsite\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     23\u001b[0m     response\u001b[38;5;241m.\u001b[39mraise_for_status()\n\u001b[0;32m     25\u001b[0m     \u001b[38;5;66;03m# Extract emails using BeautifulSoup and regex\u001b[39;00m\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\requests\\api.py:73\u001b[0m, in \u001b[0;36mget\u001b[1;34m(url, params, **kwargs)\u001b[0m\n\u001b[0;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m     63\u001b[0m     \u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[0;32m     64\u001b[0m \n\u001b[0;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[0;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m request(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mget\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, params\u001b[38;5;241m=\u001b[39mparams, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\requests\\api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[1;34m(method, url, **kwargs)\u001b[0m\n\u001b[0;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[0;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[0;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[0;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[1;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m session\u001b[38;5;241m.\u001b[39mrequest(method\u001b[38;5;241m=\u001b[39mmethod, url\u001b[38;5;241m=\u001b[39murl, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\requests\\sessions.py:587\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    582\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m    583\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[0;32m    584\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[0;32m    585\u001b[0m }\n\u001b[0;32m    586\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[1;32m--> 587\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[0;32m    589\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\requests\\sessions.py:701\u001b[0m, in \u001b[0;36mSession.send\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    698\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[0;32m    700\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[1;32m--> 701\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    703\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[0;32m    704\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\requests\\adapters.py:489\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    487\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    488\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m chunked:\n\u001b[1;32m--> 489\u001b[0m         resp \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43murlopen\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    490\u001b[0m \u001b[43m            \u001b[49m\u001b[43mmethod\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    491\u001b[0m \u001b[43m            \u001b[49m\u001b[43murl\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    492\u001b[0m \u001b[43m            \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    493\u001b[0m \u001b[43m            \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrequest\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    494\u001b[0m \u001b[43m            \u001b[49m\u001b[43mredirect\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    495\u001b[0m \u001b[43m            \u001b[49m\u001b[43massert_same_host\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    496\u001b[0m \u001b[43m            \u001b[49m\u001b[43mpreload_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    497\u001b[0m \u001b[43m            \u001b[49m\u001b[43mdecode_content\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[0;32m    498\u001b[0m \u001b[43m            \u001b[49m\u001b[43mretries\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmax_retries\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    499\u001b[0m \u001b[43m            \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    500\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    502\u001b[0m     \u001b[38;5;66;03m# Send the request.\u001b[39;00m\n\u001b[0;32m    503\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    504\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(conn, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mproxy_pool\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py:703\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    700\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_proxy(conn)\n\u001b[0;32m    702\u001b[0m \u001b[38;5;66;03m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[1;32m--> 703\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_request\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    704\u001b[0m \u001b[43m    \u001b[49m\u001b[43mconn\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    705\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmethod\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    706\u001b[0m \u001b[43m    \u001b[49m\u001b[43murl\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    707\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout_obj\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    708\u001b[0m \u001b[43m    \u001b[49m\u001b[43mbody\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbody\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    709\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheaders\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mheaders\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    710\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    711\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    713\u001b[0m \u001b[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[0;32m    714\u001b[0m \u001b[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[0;32m    715\u001b[0m \u001b[38;5;66;03m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[0;32m    716\u001b[0m \u001b[38;5;66;03m# mess.\u001b[39;00m\n\u001b[0;32m    717\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py:449\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    444\u001b[0m             httplib_response \u001b[38;5;241m=\u001b[39m conn\u001b[38;5;241m.\u001b[39mgetresponse()\n\u001b[0;32m    445\u001b[0m         \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mBaseException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    446\u001b[0m             \u001b[38;5;66;03m# Remove the TypeError from the exception chain in\u001b[39;00m\n\u001b[0;32m    447\u001b[0m             \u001b[38;5;66;03m# Python 3 (including for exceptions like SystemExit).\u001b[39;00m\n\u001b[0;32m    448\u001b[0m             \u001b[38;5;66;03m# Otherwise it looks like a bug in the code.\u001b[39;00m\n\u001b[1;32m--> 449\u001b[0m             \u001b[43msix\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mraise_from\u001b[49m\u001b[43m(\u001b[49m\u001b[43me\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m    450\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError, SocketError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    451\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mread_timeout)\n",
      "File \u001b[1;32m<string>:3\u001b[0m, in \u001b[0;36mraise_from\u001b[1;34m(value, from_value)\u001b[0m\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py:444\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    441\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m    442\u001b[0m     \u001b[38;5;66;03m# Python 3\u001b[39;00m\n\u001b[0;32m    443\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 444\u001b[0m         httplib_response \u001b[38;5;241m=\u001b[39m \u001b[43mconn\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetresponse\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    445\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mBaseException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    446\u001b[0m         \u001b[38;5;66;03m# Remove the TypeError from the exception chain in\u001b[39;00m\n\u001b[0;32m    447\u001b[0m         \u001b[38;5;66;03m# Python 3 (including for exceptions like SystemExit).\u001b[39;00m\n\u001b[0;32m    448\u001b[0m         \u001b[38;5;66;03m# Otherwise it looks like a bug in the code.\u001b[39;00m\n\u001b[0;32m    449\u001b[0m         six\u001b[38;5;241m.\u001b[39mraise_from(e, \u001b[38;5;28;01mNone\u001b[39;00m)\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\http\\client.py:1374\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1372\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m   1373\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 1374\u001b[0m         \u001b[43mresponse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbegin\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1375\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m:\n\u001b[0;32m   1376\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\http\\client.py:318\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    316\u001b[0m \u001b[38;5;66;03m# read until we get a non-100 response\u001b[39;00m\n\u001b[0;32m    317\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m--> 318\u001b[0m     version, status, reason \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_read_status\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m status \u001b[38;5;241m!=\u001b[39m CONTINUE:\n\u001b[0;32m    320\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\http\\client.py:279\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    278\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_read_status\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[1;32m--> 279\u001b[0m     line \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreadline\u001b[49m\u001b[43m(\u001b[49m\u001b[43m_MAXLINE\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124miso-8859-1\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m    280\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(line) \u001b[38;5;241m>\u001b[39m _MAXLINE:\n\u001b[0;32m    281\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m LineTooLong(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mstatus line\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32mC:\\ProgramData\\anaconda3\\lib\\socket.py:705\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    703\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[0;32m    704\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 705\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_sock\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrecv_into\u001b[49m\u001b[43m(\u001b[49m\u001b[43mb\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    706\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m timeout:\n\u001b[0;32m    707\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_timeout_occurred \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import re\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import openpyxl\n",
    "\n",
    "# Open the Excel file\n",
    "workbook = openpyxl.load_workbook('pet groomers.xlsx')\n",
    "sheet = workbook.active\n",
    "\n",
    "# Column containing the website URLs\n",
    "website_column = 'G'\n",
    "\n",
    "# Column to store the extracted emails\n",
    "email_column = 'H'\n",
    "\n",
    "# Iterate through the rows, starting from the second row\n",
    "for row in range(2, sheet.max_row + 1):\n",
    "    website = sheet[website_column + str(row)].value\n",
    "\n",
    "    try:\n",
    "        # Send a GET request to the website\n",
    "        response = requests.get(website)\n",
    "        response.raise_for_status()\n",
    "\n",
    "        # Extract emails using BeautifulSoup and regex\n",
    "        soup = BeautifulSoup(response.text, 'html.parser')\n",
    "        emails = re.findall(r'\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Z|a-z]{2,}\\b', soup.get_text())\n",
    "\n",
    "        # Save emails in the email column\n",
    "        sheet[email_column + str(row)].value = ', '.join(emails)\n",
    "\n",
    "        print(f\"Emails extracted for {website}: {', '.join(emails)}\")\n",
    "    except (requests.RequestException, ValueError) as e:\n",
    "        print(f\"Error accessing website {website}: {str(e)}\")\n",
    "\n",
    "# Save the modified Excel file\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c6d04f35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error accessing website http://www.doggroomingparlour.co.uk: 403 Client Error: Forbidden for url: http://www.doggroomingparlour.co.uk/\n",
      "No emails found\n",
      "Emails extracted for https://www.awesomepaws.org/: pageinfo@awesomepaws.orgLog\n",
      "Error accessing website http://www.pawsawhiledoggrooming.co.uk: HTTPConnectionPool(host='www.pawsawhiledoggrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EC52D10>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for http://www.lolasdoggrooming.com: lolasdoggrooming@outlook.com\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://inglebypets.co.uk: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))\n",
      "No emails found\n",
      "Error accessing website http://www.amyspets.co.uk: HTTPConnectionPool(host='www.amyspets.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F32DA80>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.woofnreadypetgrooming.co.uk: 403 Client Error: Forbidden for url: http://www.woofnreadypetgrooming.co.uk/\n",
      "Error accessing website https://callywaggiesdoggrooming.co.uk: HTTPSConnectionPool(host='callywaggiesdoggrooming.co.uk', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:997)')))\n",
      "No emails found\n",
      "Error accessing website http://www.doggiedaysteesside.com: 403 Client Error: Forbidden for url: http://www.doggiedaysteesside.com/\n",
      "Error accessing website http://www.louisasdoggroomingsalon.co.uk: 403 Client Error: Forbidden for url: http://www.louisasdoggroomingsalon.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.parkvalepetgrooming.co.uk: GROOMER@PARKVALEPETGROOMING.CO.UK\n",
      "No emails found\n",
      "Emails extracted for http://www.vanity-fur-grooming.com: info@vanity-fur-grooming.com\n",
      "Error accessing website http://www.k9houseofstyle.com: 404 Client Error: Not Found for url: https://www.k9houseofstyle.com/\n",
      "Error accessing website http://www.barkbone.com: 403 Client Error: Forbidden for url: https://www.brandbucket.com/names/barkbone?source=ext\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.place4pets.co.uk: HTTPConnectionPool(host='www.place4pets.co.uk', port=80): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x0000029F5FB31ED0>, 'Connection to www.place4pets.co.uk timed out. (connect timeout=None)'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.breedabove.co.uk: info@breedabove.co.uk, info@breedabove.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://www.woofsandwags.info/: HTTPSConnectionPool(host='www.woofsandwags.info', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5C7FA3B0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "Error accessing website http://www.tomstrims.co.uk: HTTPConnectionPool(host='www.tomstrims.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5C7F2920>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.whiskerspawsandchores.co.uk: HTTPConnectionPool(host='www.whiskerspawsandchores.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F218280>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.sageprofessionaldoggrooming.co.uk/: 650711info@sageprofessionaldoggrooming.co.ukHomeDog, touchinfo@sageprofessionaldoggrooming.co.ukHighly, 6RFinfo@sageprofessionaldoggrooming.co\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.kingsroadvets.org?utm_source=yell&utm_medium=referral&utm_campaign=yell: 406 Client Error: Not Acceptable for url: http://www.kingsroadvets.org/?utm_source=yell&utm_medium=referral&utm_campaign=yell\n",
      "Emails extracted for https://www.pinkwhiskersgrooming.co.uk/: info@pinkwhiskersgrooming.co.uk\n",
      "Error accessing website http://www.groomer-has-it.com/: HTTPConnectionPool(host='www.groomer-has-it.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F21B3A0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.prettypawfectdoggrooming.co.uk: HTTPConnectionPool(host='www.prettypawfectdoggrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F21B280>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.thedogpamperer.net: 404 Client Error: Not Found for url: http://www.thedogpamperer.net/\n",
      "Emails extracted for http://www.millionhairs.co.uk/: enquiries@millionhairs.co.uk, chris@arcimedia.co.uk, enquiries@millionhairs.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.binxsgroomingempawrium.business.site: HTTPSConnectionPool(host='www.binxsgroomingempawrium.business.site', port=443): Max retries exceeded with url: / (Caused by SSLError(CertificateError(\"hostname 'www.binxsgroomingempawrium.business.site' doesn't match either of '*.google.com', '*.appengine.google.com', '*.bdn.dev', '*.origin-test.bdn.dev', '*.cloud.google.com', '*.crowdsource.google.com', '*.datacompute.google.com', '*.google.ca', '*.google.cl', '*.google.co.in', '*.google.co.jp', '*.google.co.uk', '*.google.com.ar', '*.google.com.au', '*.google.com.br', '*.google.com.co', '*.google.com.mx', '*.google.com.tr', '*.google.com.vn', '*.google.de', '*.google.es', '*.google.fr', '*.google.hu', '*.google.it', '*.google.nl', '*.google.pl', '*.google.pt', '*.googleadapis.com', '*.googleapis.cn', '*.googlevideo.com', '*.gstatic.cn', '*.gstatic-cn.com', 'googlecnapps.cn', '*.googlecnapps.cn', 'googleapps-cn.com', '*.googleapps-cn.com', 'gkecnapps.cn', '*.gkecnapps.cn', 'googledownloads.cn', '*.googledownloads.cn', 'recaptcha.net.cn', '*.recaptcha.net.cn', 'recaptcha-cn.net', '*.recaptcha-cn.net', 'widevine.cn', '*.widevine.cn', 'ampproject.org.cn', '*.ampproject.org.cn', 'ampproject.net.cn', '*.ampproject.net.cn', 'google-analytics-cn.com', '*.google-analytics-cn.com', 'googleadservices-cn.com', '*.googleadservices-cn.com', 'googlevads-cn.com', '*.googlevads-cn.com', 'googleapis-cn.com', '*.googleapis-cn.com', 'googleoptimize-cn.com', '*.googleoptimize-cn.com', 'doubleclick-cn.net', '*.doubleclick-cn.net', '*.fls.doubleclick-cn.net', '*.g.doubleclick-cn.net', 'doubleclick.cn', '*.doubleclick.cn', '*.fls.doubleclick.cn', '*.g.doubleclick.cn', 'dartsearch-cn.net', '*.dartsearch-cn.net', 'googletraveladservices-cn.com', '*.googletraveladservices-cn.com', 'googletagservices-cn.com', '*.googletagservices-cn.com', 'googletagmanager-cn.com', '*.googletagmanager-cn.com', 'googlesyndication-cn.com', '*.googlesyndication-cn.com', '*.safeframe.googlesyndication-cn.com', 'app-measurement-cn.com', '*.app-measurement-cn.com', 'gvt1-cn.com', '*.gvt1-cn.com', 'gvt2-cn.com', '*.gvt2-cn.com', '2mdn-cn.net', '*.2mdn-cn.net', 'googleflights-cn.net', '*.googleflights-cn.net', 'admob-cn.com', '*.admob-cn.com', 'googlesandbox-cn.com', '*.googlesandbox-cn.com', '*.safenup.googlesandbox-cn.com', '*.gstatic.com', '*.metric.gstatic.com', '*.gvt1.com', '*.gcpcdn.gvt1.com', '*.gvt2.com', '*.gcp.gvt2.com', '*.url.google.com', '*.youtube-nocookie.com', '*.ytimg.com', 'android.com', '*.android.com', '*.flash.android.com', 'g.cn', '*.g.cn', 'g.co', '*.g.co', 'goo.gl', 'www.goo.gl', 'google-analytics.com', '*.google-analytics.com', 'google.com', 'googlecommerce.com', '*.googlecommerce.com', 'ggpht.cn', '*.ggpht.cn', 'urchin.com', '*.urchin.com', 'youtu.be', 'youtube.com', '*.youtube.com', 'youtubeeducation.com', '*.youtubeeducation.com', 'youtubekids.com', '*.youtubekids.com', 'yt.be', '*.yt.be', 'android.clients.google.com', 'developer.android.google.cn', 'developers.android.google.cn', 'source.android.google.cn'\")))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Emails extracted for http://www.the-grooming-room.co.uk: Groomingroommcl@gmail.com\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.smartdogsgrooming.co.uk: HTTPConnectionPool(host='www.smartdogsgrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F67AB00>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://bellpetservices.co.uk/: 404 Client Error: Not Found for url: https://bellpetservices.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.a1prettypaws.com: HTTPConnectionPool(host='www.a1prettypaws.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EECA8F0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "Emails extracted for http://www.pawfectstyle.org.uk: enquiries@pawfectstyle.org.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.pets2impress.com: mail@pets2impress.com\n",
      "Error accessing website http://www.bellasdoggrooming.co.uk/?utm_source=yell&utm_medium=referral&utm_campaign=yell: 403 Client Error: Forbidden for url: http://www.bellasdoggrooming.co.uk/?utm_source=yell&utm_medium=referral&utm_campaign=yell\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.headtopaw.co.uk/: HTTPConnectionPool(host='www.headtopaw.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F5CB910>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "Error accessing website http://www.doggroomergateshead.co.uk: HTTPConnectionPool(host='www.doggroomergateshead.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EDEC580>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://saltwellparkdoggroomers.com/: HTTPConnectionPool(host='saltwellparkdoggroomers.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EDEC6A0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "Error accessing website https://www.k9room.co.uk: HTTPSConnectionPool(host='www.k9room.co.uk', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5EDEFFD0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.k9creche-ne.co.uk: 406 Client Error: Not Acceptable for url: http://www.k9creche-ne.co.uk/\n",
      "No emails found\n",
      "Emails extracted for https://www.pettaxiservices.co.uk?utm_source=yell&utm_medium=referral&utm_campaign=yell: pettaxiservices@outlook.com\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for https://www.edsshearings.com: edsshearings@gmail.com\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.brunosbuddiesdogwalking.co.uk/: brunosbuddies@yahoo.co.uk, brunosbuddies@yahoo.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://petnannyest1998.co.uk?utm_source=yell&utm_medium=referral&utm_campaign=yell: HTTPSConnectionPool(host='petnannyest1998.co.uk', port=443): Max retries exceeded with url: /?utm_source=yell&utm_medium=referral&utm_campaign=yell (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000029F5DAD84F0>, 'Connection to petnannyest1998.co.uk timed out. (connect timeout=None)'))\n",
      "Error accessing website http://www.tatsandtails.com: HTTPConnectionPool(host='www.tatsandtails.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5DAFD5A0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for https://www.deandoggies.co.uk/: items.info@deandoggies.co, Doggiesinfo@deandoggies.co\n",
      "Error accessing website http://www.preciouspamperedpooches.co.uk: HTTPConnectionPool(host='www.preciouspamperedpooches.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5ECC3070>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for https://www.oscars.co.uk/branches/darlington: gill.innes@oscars.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.pupparel.co.uk?utm_source=yell&utm_medium=referral&utm_campaign=yell: info@pupparel.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://ellehcimpaceyourpooch.co.uk/: ellehcimpyp@gmail.com\n",
      "No emails found\n",
      "Emails extracted for http://www.millbryhill.co.uk: 5NZsales@millbryhill.co\n",
      "Emails extracted for http://www.millbryhill.co.uk/: 5NZsales@millbryhill.co\n",
      "No emails found\n",
      "Error accessing website http://www.whitecrossvets.co.uk/our-surgeries/coulby-newham/: 403 Client Error: Forbidden for url: http://www.whitecrossvets.co.uk/our-surgeries/coulby-newham/\n",
      "No emails found\n",
      "Error accessing website http://www.grangevets.co.uk: 403 Client Error: Forbidden for url: http://www.grangevets.co.uk/\n",
      "Emails extracted for http://www.aristocatsandallsorts.co.uk?utm_source=yell&utm_medium=referral&utm_campaign=yell: jgillespie237@btinternet.com, jgillespie237@btinternet.com\n",
      "Emails extracted for http://www.coasttocoast.co.uk: info@coasttocoast.co.uk, info@coasttocoast.co.uk\n",
      "Emails extracted for http://www.hurworthburnpethotel.co.uk: 838336contact@hurworthburnkennels.co.uk, contact@hurworthburnkennels.co.ukFind\n",
      "Error accessing website http://www.clifforlodgevets.co.uk: HTTPConnectionPool(host='www.clifforlodgevets.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F3C75B0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.blueacres.co.uk: HTTPConnectionPool(host='www.blueacres.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F94F370>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for http://www.groomsbykaren.com: 659enquiries@groomsbykaren.com, enquiries@groomsbykaren.com\n",
      "No emails found\n",
      "Emails extracted for http://www.greencroftdogdaycare.co.uk/: david.lowes@greencroftdogdaycare.co.ukSend, david.lowes@greencroftdogdaycare.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.karenscaninecutsderby.co.uk/: 403 Client Error: Forbidden for url: http://www.karenscaninecutsderby.co.uk/\n",
      "No emails found\n",
      "Error accessing website http://www.thefurdressers.com: HTTPConnectionPool(host='www.thefurdressers.com', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5E355A50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.wilsonvetgroup.co.uk: 403 Client Error: Forbidden for url: https://www.wilsonvetgroup.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.durhampetcare.com?utm_source=yell&utm_medium=referral&utm_campaign=yell: durhampetcare@sky.com, durhampetcare@sky.com, durhampetcare@sky.com, durhampetcare@sky.com\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No emails found\n",
      "Emails extracted for http://www.skyecroft.co.uk: enquiries@skyecroft.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.4pawsfarm.co.uk: 403 Client Error: Forbidden for url: http://www.4pawsfarm.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.princebishopvets.co.uk: 403 Client Error: Forbidden for url: https://www.princebishopvets.co.uk/\n",
      "No emails found\n",
      "Emails extracted for http://www.lanchesterpetresort.co.uk: info@lanchesterpetresort.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.petsathome.com/: 403 Client Error: Forbidden for url: http://www.petsathome.com/\n",
      "Error accessing website http://www.bowwowbeauty.co.uk: HTTPSConnectionPool(host='www.bowwowbeauty.co.uk', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:997)')))\n",
      "Error accessing website http://mutz-n-moggies.co.uk: HTTPConnectionPool(host='mutz-n-moggies.co.uk', port=80): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x0000029F5EBB5780>, 'Connection to mutz-n-moggies.co.uk timed out. (connect timeout=None)'))\n",
      "No emails found\n",
      "Error accessing website http://www.muckypups-org.co.uk: 403 Client Error: Forbidden for url: http://www.muckypups-org.co.uk/\n",
      "No emails found\n",
      "Error accessing website http://www.cosyposts.com/: HTTPConnectionPool(host='www.cosyposts.com', port=80): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x0000029F5F289900>, 'Connection to www.cosyposts.com timed out. (connect timeout=None)'))\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://www.purrfectionbyisobel.co.uk?utm_source=yell&utm_medium=referral&utm_campaign=yell: HTTPSConnectionPool(host='www.purrfectionbyisobel.co.uk', port=443): Max retries exceeded with url: /?utm_source=yell&utm_medium=referral&utm_campaign=yell (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5FA65AB0>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed'))\n",
      "No emails found\n",
      "Error accessing website http://www.doggiestyleleeds.co.uk: 403 Client Error: Forbidden for url: http://www.doggiestyleleeds.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.alwoodleydoggrooming.com: info@alwoodleydoggrooming.com, info@alwoodleydoggrooming.com, info@alwoodleydoggrooming.com\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.fortheloveofcatsanddogs.co.uk: info@fortheloveofcatsanddogs.co.uk\n",
      "No emails found\n",
      "Emails extracted for https://mrsbsdogs.co.uk: info@mrsbsdogs.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://www.classycuts.org.uk: HTTPSConnectionPool(host='www.classycuts.org.uk', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5ECC18D0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Emails extracted for https://www.laburnumboardingkennels.co.uk/: info@laburnumboardingkennels.co.uk, info@laburnumboardingkennels.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.dogs-aloud.co.uk/: 403 Client Error: Forbidden for url: https://www.dogs-aloud.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.headsntailsdoggrooming.co.uk: carol_watt@yahoo.co.uk, carol_watt@yahoo.co.uk\n",
      "No emails found\n",
      "Error accessing website http://www.thegrooming-room.co.uk/: 403 Client Error: Forbidden for url: http://www.thegrooming-room.co.uk/\n",
      "Emails extracted for http://www.rufftofluffgrooming.co.uk: info@rufftofluffgrooming.co.uk\n",
      "No emails found\n",
      "Error accessing website http://www.ardsley-boarding-kennels.co.uk/: 403 Client Error: Forbidden for url: http://www.ardsley-boarding-kennels.co.uk/\n",
      "Emails extracted for http://www.preciouspawsleeds.co.uk/: precious-paws@hotmail.co.uk\n",
      "No emails found\n",
      "Emails extracted for http://www.cuddlesandclips.com: RealTime@gname.com\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.animalpla.net: HTTPConnectionPool(host='www.animalpla.net', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F495570>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "Error accessing website http://www.poodledoos.co.uk: HTTPConnectionPool(host='www.poodledoos.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F48DE70>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://muckypawsgrooming.co.uk/: HTTPConnectionPool(host='muckypawsgrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F48DD50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.muddypawz.me.uk: HTTPConnectionPool(host='www.muddypawz.me.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5F48DD50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.campusvetsandpets.co.uk: HTTPSConnectionPool(host='www.campusvetsandpets.co.uk', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x0000029F5F48D9C0>, 'Connection to www.campusvetsandpets.co.uk timed out. (connect timeout=None)'))\n",
      "Emails extracted for http://www.therhyddings.co.uk: rhyddingskennels@gmail.com, rhyddingskennels@gmail.com\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.preciouspawsleeds.co.uk: precious-paws@hotmail.co.uk\n",
      "Emails extracted for https://www.jandrpetservices.co.uk/: info@jandrpetservices.co\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.harenhounds.info/: HTTPConnectionPool(host='www.harenhounds.info', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EF7AC50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://www.thebarkinglot.co.uk/: 406 Client Error: Not Acceptable for url: http://www.thebarkinglot.co.uk/\n",
      "Error accessing website http://www.doggydazzler.wixsite.com/dogs: HTTPSConnectionPool(host='www.doggydazzler.wixsite.com', port=443): Max retries exceeded with url: /dogs (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:997)')))\n",
      "Error accessing website http://www.chicandneatdoggrooming.co.uk: HTTPConnectionPool(host='www.chicandneatdoggrooming.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5EF1FAC0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://stride-and-groom.co.uk/: info@stride-and-groom.co.uk\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website https://www.luckysleads.co.uk/?utm_source=yell&utm_medium=referral&utm_campaign=yell: HTTPSConnectionPool(host='www.luckysleads.co.uk', port=443): Max retries exceeded with url: /?utm_source=yell&utm_medium=referral&utm_campaign=yell (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000029F5F287310>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Error accessing website http://stsuppliesonline.co.uk: HTTPConnectionPool(host='stsuppliesonline.co.uk', port=80): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x0000029F5DEF28C0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))\n",
      "Error accessing website http://www.coop.co.uk: 403 Client Error: Forbidden for url: https://www.coop.co.uk/\n",
      "No emails found\n",
      "No emails found\n",
      "No emails found\n",
      "Emails extracted for http://www.hillamfeeds.co.uk: tackagain@aol.com\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "import openpyxl\n",
    "\n",
    "# Open the Excel file\n",
    "workbook = openpyxl.load_workbook('pet groomers.xlsx')\n",
    "sheet = workbook.active\n",
    "\n",
    "# Column containing the website URLs\n",
    "website_column = 'G'\n",
    "\n",
    "# Column to store the extracted emails\n",
    "email_column = 'H'\n",
    "# Iterate through the rows, starting from the second row\n",
    "for row in range(2, sheet.max_row + 1):\n",
    "    website = sheet[website_column + str(row)].value\n",
    "\n",
    "    try:\n",
    "        # Send a GET request to the website\n",
    "        response = requests.get(website)\n",
    "        response.raise_for_status()\n",
    "\n",
    "        # Extract emails using BeautifulSoup and regex\n",
    "        soup = BeautifulSoup(response.text, 'html.parser')\n",
    "        emails = re.findall(r'\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Z|a-z]{2,}\\b', soup.get_text())\n",
    "\n",
    "        # Save emails in the email column or empty string if not found\n",
    "        sheet[email_column + str(row)].value = ', '.join(emails) if emails else ''\n",
    "\n",
    "        print(f\"Emails extracted for {website}: {', '.join(emails)}\" if emails else \"No emails found\")\n",
    "    except (requests.RequestException, ValueError) as e:\n",
    "        print(f\"Error accessing website {website}: {str(e)}\")\n",
    "\n",
    "# Save the modified Excel file\n",
    "workbook.save('output.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3819a9bd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
